samples/batch-process-document.v1beta3.js

```diff
@@ -0,0 +1,163 @@
+
+    // TODO: Remove try/catch block when appropriate changes are made to service.
+    try {
+      const [operation] = await client.batchProcessDocuments(request);
+
+      // Wait for operation to complete.
+      await operation.promise();
+    } catch (ex) {
+      // CURRENTLY the LRO returned from service has 'done = true' but
+      // does not provide a result. This raises an error from the google-gax
+      // library, despite the LRO actually succeeeding.
+      const lroError =
+        'Long running operation has finished but there was no result';
+
+      if (ex.message.indexOf(lroError) === -1) {
+        throw ex;
+      }
+    }
+
+    console.log('Document processing complete.');
+
+    // Query Storage bucket for the results file(s).
+    const query = {
+      prefix: gcsOutputUriPrefix,
+    };
+
+    console.log('Fetching results ...');
+
+    // List all of the files in the Storage bucket
+    const [files] = await storage.bucket(gcsOutputUri).getFiles(query);
+
+    // Add all asynchronous downloads to queue for execution.
+    const queue = new PQueue({concurrency: 15});
+    const tasks = files.map((fileInfo, index) => async () => {
+      // Get the file as a buffer
+      const [file] = await fileInfo.download();
+
+      console.log(`Fetched file #${index + 1}:`);
+
+      // The results stored in the output Storage location
+      // are formatted as a document object.
+      const document = JSON.parse(file.toString());
+      const {text} = document;
+
+      // Extract shards from the text field
+      const getText = textAnchor => {
+        if (!textAnchor.textSegments || textAnchor.textSegments.length === 0) {
+          return '';
+        }
+
+        // First shard in document doesn't have startIndex property
+        const startIndex = textAnchor.textSegments[0].startIndex || 0;
+        const endIndex = textAnchor.textSegments[0].endIndex;
+
+        return text.substring(startIndex, endIndex);
+      };
+
+      // Read the text recognition output from the processor
+      console.log('The document contains the following paragraphs:');
+
+      const [page1] = document.pages;
+      const {paragraphs} = page1;
+      for (const paragraph of paragraphs) {
+        const paragraphText = getText(paragraph.layout.textAnchor);
+        console.log(`Paragraph text:\n${paragraphText}`);
+      }
+
+      // Form parsing provides additional output about
+      // form-formatted PDFs. You  must create a form
+      // processor in the Cloud Console to see full field details.
+      console.log('\nThe following form key/value pairs were detected:');
+
+      const {formFields} = page1;
+      for (const field of formFields) {
+        const fieldName = getText(field.fieldName.textAnchor);
+        const fieldValue = getText(field.fieldValue.textAnchor);
+
+        console.log('Extracted key value pair:');
+        console.log(`\t(${fieldName}, ${fieldValue})`);
+      }
+    });
+    queue.addAll(tasks);
```

Comment:
I believe you need to `await` this, or the function will exit before the work is all done. Is the result of `addAll` a promise, I think?

Reply:
Done.